src/main/java/picoded/core/conv/MapValueConv.java

```diff
@@ -88,80 +88,58 @@ protected MapValueConv() {
 		
 		return finalMap;
 	}
-	
-	@SuppressWarnings("unchecked")
-	public static Map<String, Object> toFullyQualifiedKeys(Object source, String rootName,
-		String separator) {
-		Map<String, Object> fullyQualifiedMap = new HashMap<String, Object>();
-		
-		if (rootName == null) {
-			rootName = "";
-		}
-		
-		if (separator.isEmpty()) {
-			separator = ".";
-		}
-		String parentName;
-		if (source instanceof List) {
-			List<Object> sourceList = (List<Object>) source;
-			
-			int counter = 0;
-			for (Object obj : sourceList) {
-				parentName = getRootName(rootName, counter);
-				if (obj instanceof List) {
-					fullyQualifiedMap.putAll(toFullyQualifiedKeys(obj, parentName, separator));
-					++counter;
-					
-				}
-				if (obj instanceof Map) {
-					Map<String, Object> objMap = (Map<String, Object>) obj;
-					fullyQualifiedMap = getFullyQualifiedMap(fullyQualifiedMap, objMap, rootName,
-						parentName, counter, separator);
-					counter = (int) fullyQualifiedMap.get("counter");
-				}
-			}
-		} else if (source instanceof Map) {
+
+	/**
+	 * Recursively loop through the source object to form the absolute path to every single item and place them
+	 * inside a Map<String, Object> to return back to the user.
+	 *
+	 * @param source The source to loop through
+	 * @param parentName The current parent path
+	 * @param delimiter The delimiter to seperate each level
+	 * @return a Map of absolute paths
+	 */
+	public static Map<String, Object> convertToFullyQualifyNames(Object source, String parentName, String delimiter){
```

Comment:
Having this named as `MapValueConv.toFullyQualifiedNames` instead of `MapValueConv.convertToFullyQualifyNames` makes more sense as its a static class.